plugin/pkg/shoot/dnshostedzone/admission.go

```diff
@@ -135,6 +137,15 @@ func (d *DNSHostedZone) Admit(a admission.Attributes) error {
 		return nil
 	}
 
+	// Alicloud DNS needs no HostedZoneID
+	if shoot.Spec.DNS.Provider == garden.DNSAlicloud {
+		shoot.Spec.DNS.HostedZoneID = nil
```

Comment:
We should not set it explicitly to `nil` here (you already have validated in the static validation in `pkg/apis/garden/validation`) that it is `nil`.

Reply:
I found some potential issues of addmission.go and will follow up.